highlighters/main/main-highlighter.zsh

```diff
@@ -563,12 +645,15 @@ _zsh_highlight_highlighter_main_paint()
                           esac
                           if (( insane_alias )); then
                             style=unknown-token
+                          # Calling 'type' again; since __type memoizes the answer, this call is just a hash lookup.
+                          elif _zsh_highlight_main__type "$arg" && [[ $REPLY == 'none' ]]; then
```

Comment:
Should this also be __is_runnable?

Reply:
Could, and check for $? = 1. Not sure whether it'd be clearer.

Reply:
No need for the explicit `$? = 1`, `elif _zsh_highlight_main__is_runnable; then` is sufficient.

Reply:
[summary of IRC]

What should this do when `_zsh_highlight_main__type` fails (returns `$? = 1`)? Currently it falls back to the `alias` block below. Should it return `unknown-token`? Or maybe some new `fallback` style (since this isn't a syntactically-invalid BUFFER;, it's a "we don't know whether the command line is syntactically valid" situation)? In the latter case, should `$style` be initialised to `fallback` rather than `unknown-token`? (That would ripple to the opening condition of the non-command-word branch.)

Reply:
I think for now it'd be acceptable to `s/&&/||/` here and then open a new issue on `fallback`/`unknown-token`.